src/Tuura/Plato/Translation.hs

```diff
@@ -89,21 +89,27 @@ validateInterface signs circuit
 
 -- Perform cartesian product on list of lists. This will also sort and remove
 -- duplicates in sublists, and remove supersets for the most compact form.
-cartesianProduct :: Ord a => NonEmpty.NonEmpty [a] -> [[a]]
+cartesianProduct :: Ord a => NonEmpty.NonEmpty [Transition a] -> [[Transition a]]
 cartesianProduct l = removeSupersets sortAllLists
```

Comment:
I would simplify this as `removeSupersets $ map (sort . removeRedundancies . nub) sequenced`.